msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java

```diff
@@ -209,9 +249,10 @@ public JsonToken nextToken()
                 }
                 break;
             case FLOAT:
-                value = messageUnpacker.unpackValue(var);
+                type = Type.DOUBLE;
+                doubleValue = messageUnpacker.unpackDouble();
```

Comment:
float to double conversion has some overhead. If you want to make it faster, consider to split reading float/double values separately (similarly int/long) in future.